modules/search/class.jetpack-search.php

```diff
@@ -216,10 +216,42 @@ public function load_assets() {
 				$script_version = self::get_asset_version( $script_relative_path );
 				$script_path    = plugins_url( $script_relative_path, JETPACK__PLUGIN_FILE );
 				wp_enqueue_script( 'jetpack-instant-search', $script_path, array(), $script_version, true );
-				$_blog_id = Jetpack::get_option( 'id' );
+				$this->load_and_initialize_tracks();
```

Comment:
I am curious about this. Does this mean we'll now track and log searches from any visitor, as soon as the Jetpack master owner has connected the site to WordPress.com and thus implicitly accepted our ToS? Is that alright? What happens if that master owner has opted out using the Privacy settings at `wp-admin/admin.php?page=jetpack#/privacy`?

In relation with the GDPR, what happens if a specific user requests that all their search logs be deleted? Is that something that the site owner would have to comply to, and how would they go about this?

Reply:
> What happens if that master owner has opted out using the Privacy settings at `wp-admin/admin.php?page=jetpack#/privacy`?

For those reading along, the Privacy Setting has a toggle control with the following text:

> Share information with our analytics tool about your use of services while logged in to your WordPress.com account.

In all honesty, I feel uneasy with our current implementation. Yes, the toggle there exists to satisfy the letter of the Cookie Law. Since Instant Search (ideally) won't handle user data via cookies, we technically won't need to respect the setting there.

However, it's important to note that we will begin tracking *all* visitors to any Jetpack site with Instant Search enabled with zero recourse for opting out.

I think we really ought to respect this site option for user analytics in Instant Search.

@bperson, your thoughts on this would be appreciated.

Reply:
Marking this as resolved since this is being tracked and discussed in #13762